### DISCUSSION TITLE
Storage Egress Counting

### DISCUSSION THREAD

**Discussion by furkananter** [supabase/supabase#14316] [Questions] [answered]
Hi, I've developed profile photo feature for my application and I especially created that feature with download:false. Example down below: I am so much careful about storage egress because this is basically social media application and its growing fast. So we need to care about architecture and scalability. Why this is happening. I am ensuring the code:

``` jsx
const {data} = supabase
.storage
.from('profilePhotos')
.getPublicUrl(fileName, {
download: false,
});
```
My code working with version system. It means: When user upload their new profile photo to their row, I am adding new version count. Count is at the end of my publicUrl code. like: ?version=22, so I can run away from cache.

When I get this feature live, It was just 995 mb. 995 mb is just my fault. Now it is 1.1GB. It doesn't matter right now. But I am wondering why this is counting as a download when is already false? Have a good day. I am waiting your response.

**Comment by GaryAustin1** [comment]
If you use the publicURL in a img tag that will count as storage egress (I'm pretty sure).  
Depending on how you are cache busing that will also cause the browser not to cache the image.  So hopefully you only change that on a new file upload and not on a page reload.

**Reply by furkananter** [reply]
No no, I am just changing the version when users upload a new photo. My function is like that:
```jsx

  const uploadAvatarToSupabase = async (
    base64FileData: string,
    userId: string,
  ) => {
    try {
      const fileExtension = 'jpeg';
      const fileName = `${usersId}.${fileExtension}`;

      const {error} = await supabase.storage
        .from('bucket')
        .upload(fileName, decode(base64FileData), {
          upsert: true,
          contentType: 'image/jpeg',
        });

      if (error) {
        if (error.message.includes('maximum allowed size')) {
          Alert.alert('Alert');
        }
      } else {
        const {data} = supabase.storage
          .from('profilePhotos')
          .getPublicUrl(fileName, {
            download: false,
          });

        const userProfile = await supabase
          .from('profiletables')
          .select('avatar_versions')
          .eq('id', usersId)
          .single();

        const newVersion = (userProfile?.data?.avatar_version || 0) + 1;

        await supabase
          .from('profiles')
          .update({avatar_versions: newVersion})
          .eq('id', usersId);

        setAvatarUrl(`${data?.publicUrl}?version=${newVersion}`);
        setUploading(false);
      }
    } catch (error) {
      Alert.alert('Hata');
      setAvatarUrl('');
      setUploading(false);
    }
  };

```
What do you think, is that a bad solution? I believe that i did everything correctly.  Do you suggest anything else for optimization?

**Reply by GaryAustin1** [reply, marked as answer]
One thing you might consider is making your cache time much longer.
https://supabase.com/docs/guides/storage/cdn#cache-duration
Since you change the version when the image changes you really don't need the file be reloaded in the browser after 1 hour (the default).
```
.upload('public/avatar1.png', avatarFile, {
    cacheControl: '2500000',
    })
```
That is about a month of caching.
This assumes that your url with the version gets used where ever the avatar is referenced.

**Reply by furkananter** [reply]
oh, okay I'll do it. Thank you very much Gary!
